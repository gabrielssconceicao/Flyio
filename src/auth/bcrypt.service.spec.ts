import * as bcrypt from 'bcryptjs';
import { BcryptService } from './bcrypt.service';

jest.mock('bcryptjs');

describe('BcryptService', () => {
  let bcryptService: BcryptService;

  beforeEach(() => {
    bcryptService = new BcryptService();
  });

  describe('<Hash />', () => {
    it('should generate a hashed password', async () => {
      const password = 'securePassword';
      const hashedPassword = 'hashedSecurePassword';

      // Mocking bcrypt functions
      (bcrypt.genSalt as jest.Mock).mockResolvedValue('salt');
      (bcrypt.hash as jest.Mock).mockResolvedValue(hashedPassword);

      const result = await bcryptService.hash(password);

      expect(bcrypt.genSalt).toHaveBeenCalled();
      expect(bcrypt.hash).toHaveBeenCalledWith(password, 'salt');
      expect(result).toBe(hashedPassword);
      expect(result).toMatchSnapshot(password);
    });
  });

  describe('<Compare />', () => {
    it('should return true if passwords match', async () => {
      const password = 'securePassword';
      const hash = 'hashedSecurePassword';

      // Mocking bcrypt compare function
      (bcrypt.compare as jest.Mock).mockResolvedValue(true);

      const result = await bcryptService.compare(password, hash);

      expect(bcrypt.compare).toHaveBeenCalledWith(password, hash);
      expect(result).toBe(true);
      expect(result).toMatchSnapshot(password);
    });

    it('should return false if passwords do not match', async () => {
      const password = 'securePassword';
      const hash = 'wrongHashedPassword';

      // Mocking bcrypt compare function
      (bcrypt.compare as jest.Mock).mockResolvedValue(false);

      const result = await bcryptService.compare(password, hash);

      expect(bcrypt.compare).toHaveBeenCalledWith(password, hash);
      expect(result).toBe(false);
      expect(result).toMatchSnapshot(password);
    });
  });
});
