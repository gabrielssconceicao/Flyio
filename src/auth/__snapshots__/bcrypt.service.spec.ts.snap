// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BcryptService <Compare /> should return false if passwords do not match: securePassword 1`] = `false`;

exports[`BcryptService <Compare /> should return true if passwords match: securePassword 1`] = `true`;

exports[`BcryptService <Hash /> should generate a hashed password: securePassword 1`] = `"hashedSecurePassword"`;
